### PR TITLE
Precursor verification documentation

### PR DESCRIPTION
**Pull request by spdomin**
* New section for precursor verification. I outlined the general use case
in the context of an open-jet simulation study that SNL is running in
partnership with Stanford PSAAP-2.

* Added sub-section on the precursor simulation verification. Towards this
end, I ran a CTV verification problem using a precursor refined CTV
simulation. This data was used for the refinement study. Results indicate
second order accuracy - as expected - when pulling the data from a
external_field_provider realm.

* Added a section for the ABL use case. Demonstrated precursor and subsequent
simulation run with recycling of the precursor data.

* Code change to allow shell-like 2D elements for the CTV use case.

Notes:

a) Verification test is captured in the ABL use case. I probably will not add
the CTV since it is duplicated in nature.

### CODE REVIEW

File: docs/source/verification/verification.rst [hunk 1]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+different time steps unless an overly restrictive rule is enforced, i.e., a fixed timestep
+for each simulation. 
+
+In order to support such use cases, extensive usage of the the Sierra Toolkit infrastructure 
+is expected, most notably within the IO and Transfer modules. The IO module can be used to interpolate 
+the precursor simulation boundary data to the appropriate time required by the subsequent 
+simulation. Specifically, the IO module linearly interpolates between the closest data interval in the
+precursor data set. A recycling offset factor is included within the IO interface that allows
+for the cycling of data over the full time scale of interest within the subsequent simulation. For 
+typical statistically stationary turbulent flows, this is be useful to ensure proper statistics
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: "this is useful"

File: docs/source/verification/verification.rst [hunk 2]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+are captured in subsequent runs.
+
+After the transient data set from the precursor simulation is interpolated to the proper time, 
+the data is spatially interpolated and transferred to the subsequent simulation mesh using the 
+STK Transfer module. Efficient coarse parallel searches (point/bounding box) provide the list of 
+candidate owning elements on which the fine-scale search operates to determine the best search 
+candidate. The order of spatial interpolation depends on the activated numerical discretization. 
+Therefore, by combining the two STK modules, the end use case to support suck data transfers of
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: "suck?"  "stk?"

File: docs/source/verification/verification.rst [hunk 3]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+boundary data is supported.
+
+As noted, there are many other use cases in addition to the overviewed turbulent jet simulation
+that requires such temporal/spatial interpolation capabilities. For example, in typical wind
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: "that require"

File: docs/source/verification/verification.rst [hunk 4]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+
+As noted, there are many other use cases in addition to the overviewed turbulent jet simulation
+that requires such temporal/spatial interpolation capabilities. For example, in typical wind 
+farm simulation applications, a proper atmospheric boundary layer (ABL) configuration is required
+to capture a given energy state of the boundary layer. In this case, a periodic precusor ABL is run 
+with the intent of providing the inlet condition to the subsequent wind farm domain. As with the 
+previous description, the infrustructure requirements remain the same.
+
+Finally, the general creation of an "input_output" region can be useful in validation cases
+where data is provided at a subset of the overall simulation domain. Such is the case in PIV and
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: "data are provided"

File: docs/source/verification/verification.rst [hunk 5]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+.. figure:: figures/U_x_precursor.pdf
+   :width: 500px
+   :align: center
+
+   Temporal plot of the exact x-velocity component and precursor output.
+
+With the precursor data base containing the full transient data, a refinement
+study can be accomplished to determine numerical errors. Although the full machinery
+for temporal and spatial interpolation is active, the data requirement at the coarse simulations
+are represented as the subsets of the full data - both is space and time. As such, no numerical
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: "in space and time"

File: docs/source/verification/verification.rst [hunk 6]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+polution of second-order accuracy is expected. The subsequent simulations are run with an
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: degradation rather than pollution

File: docs/source/verification/verification.rst [hunk 7]
```diff
@@ -460,3 +460,178 @@ patch test was run. The temperature solution was simply,
...
+
+   Temporal accuracy plot of the x-velocity component norms using the precursor data.
+
+
+Application Verification Test; ABL Precursor/Subsequent
++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+The second, and final application test is an ABL-based simulation that first runs a precursor periodic
+solution in order to capture an appropriate ABL specification. The boundary data saved from the precursor
+simualtion is then used as an inflow boundary condition for the subsequent ABL simulation. As the precurosr
```
Comment (mbarone81) [inline docs/source/verification/verification.rst]: sp. simulation, and "are then used"

**Comment by spdomin** [comment]
Many thanks... I have made all of the corrections and will re-commit...

**Pull request merged**